### PR TITLE
DLC: Two Change SPKs

### PR DESCRIPTION
**Pull request by nkohen**
Added change spk so both local and remote have one

### CODE REVIEW

File: dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelf.scala [hunk 1]
```diff
@@ -150,15 +153,22 @@ case class BinaryOutcomeDLCWithSelf(
...
+  private val remoteFunding =
+    remoteFundingUtxos.foldLeft(0L)(_ + _.amount.satoshis.toLong)
+
   /** Remote's payout in the Win case (in which Remote loses) */
   val remoteWinPayout: CurrencyUnit = totalInput - localWinPayout
 
   /** Remote's payout in the Lose case (in which Remote wins) */
   val remoteLosePayout: CurrencyUnit = totalInput - localLosePayout
 
+  private val emptyChangeSPK: ScriptPubKey = EmptyScriptPubKey
```
Comment (Christewart) [inline dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelf.scala:175]: Why is this needed now?

File: dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelf.scala [hunk 2]
```diff
@@ -172,12 +182,20 @@ case class BinaryOutcomeDLCWithSelf(
...
+      TransactionOutput(Satoshis(localFunding) - localInput, localChangeSPK)
+    val remoteChange =
+      TransactionOutput(Satoshis(remoteFunding) - remoteInput, remoteChangeSPK)
 
-    val outputs: Vector[TransactionOutput] = Vector(output)
+    val outputs: Vector[TransactionOutput] =
+      Vector(output, localChange, remoteChange)
     val txBuilderF: Future[BitcoinTxBuilder] =
-      BitcoinTxBuilder(outputs, fundingUtxos, feeRate, changeSPK, network)
+      BitcoinTxBuilder(outputs, fundingUtxos, feeRate, emptyChangeSPK, network)
```
Comment (Christewart) [inline dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelf.scala:198]: So I think i understand why this is needed, but it seems like the better way to handle this is with an `Option` type where the `TxBuilder` fails if there is change. If we have a bug where we send money to the empty changeSPK that means that money is lost forever. This is probably something else that should be addressed in #897 

One way I would maybe feel a bit more comfortable with this is if we use the [`sign()`](https://github.com/bitcoin-s/bitcoin-s/blob/master/core/src/main/scala/org/bitcoins/core/wallet/builder/TxBuilder.scala#L216) method that allows you specify an invariant about the resulting transaction. The invariant in this case would be that we do NOT have an output that is the empty SPK.

File: dlc/src/test/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelfIntegrationTest.scala [hunk 3]
```diff
@@ -189,7 +191,8 @@ class BinaryOutcomeDLCWithSelfIntegrationTest extends BitcoindRpcTest {
         localLosePayout = localInput - CurrencyUnits.oneMBTC,
         timeout = tomorrowInBlocks,
         feeRate = feeRate,
-        changeSPK = changeSPK,
+        localChangeSPK = localChangeSPK,
+        remoteChangeSPK = remoteChangeSPK,
         network = RegTest
```
Comment (Christewart) [inline dlc/src/test/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelfIntegrationTest.scala:196]: We probably should have assertions in this test case that the resulting signed tx does not contain the empty spk. Same with `BinaryOutcomeDLCWithSelfTest`

File: dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelf.scala (hunk 2 above)
Comment (nkohen) [reply dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelf.scala:198]: Added the invariant for now and opened #961 for later
